api/apiToken/ApiTokenRestHandler.go

```diff
@@ -217,3 +218,36 @@ func (handler ApiTokenRestHandlerImpl) checkManagerAuth(token string, object str
 	}
 	return true
 }
+
+func (impl ApiTokenRestHandlerImpl) GetAllApiTokensForWebhook(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+
+	// handle super-admin RBAC
+	v := r.URL.Query()
+	projectName := v.Get("projectName")
+	environmentName := v.Get("environmentName")
+	appName := v.Get("appName")
+
+	// service call
+	res, err := impl.apiTokenService.GetAllApiTokensForWebhook(projectName, environmentName, appName, impl.CheckAuthorizationForWebhook)
+	if err != nil {
+		impl.logger.Errorw("service err, GetAllApiTokensForWebhook", "err", err)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	common.WriteJsonResp(w, err, res, http.StatusOK)
```

Comment:
we are not checking authorization of api tokens?

Reply:
yes we are checking, passing auth function to the service.